File: routes/user/ordlist.js
var cates = require("./models/cates.js");
var ordinfos = require("./models/ordinfos.js");
var res = require('./lib/resp.js');
var session = require('./lib/session.js');
var Common = require('./lib/Common.js');


var user = session.getUser();
if (!user) {
  return res.redirect('/routes/user/login.js');
}

if (request.getMethod() === 'GET') {
  var allData = cates.find();
  var data = {
    tree: cates.getTree(allData),
    user: user,
    cart: session.getCart(),
    ords: ordinfos.find({ userId: user.oid }, { sort: { id: 'desc' } }),
    toDateStr: Common.toDateStr,
  };
  res.render('./views/user/ordlist.ejs', data);
}

File: routes/user/ordlist.go
package user

import (
	"github.com/zengming00/go-testShop/framework"
	"github.com/zengming00/go-testShop/lib"
)

func Ordlist(ctx *framework.HandlerContext) {
	var user = ctx.GetUser()
	if user == nil {
		ctx.Redirect("/routes/user/login.go")
		return
	}
	if ctx.R.Method == "GET" {
		var allData = ctx.Cates.Find()
		var data = map[string]interface{}{
			"tree": ctx.Cates.GetTree(allData),
			"user": user,
			"cart": ctx.GetCart(),
			"ords": ctx.Ordinfos.Find(
				map[string]interface{}{"userId": user.Oid},
				map[string]interface{}{
					"sort": map[string]interface{}{"id": "desc"},
				},
			),
		}
		ctx.Render("./views/user/ordlist.html", data, lib.FuncMap)
		return
	}
}
